config/airflow_local_settings.py

```diff
@@ -1,11 +1,47 @@
+from copy import deepcopy
+
+from kubernetes.client import models as k8s
+from kubernetes.client.models import V1Container, V1Pod
+
 STATE_COLORS = {
-    "queued": 'gray',
-    "running": 'lime',
-    "success": '#0000FF',
-    "failed": 'red',
-    "up_for_retry": 'gold',
-    "up_for_reschedule": 'turquoise',
-    "upstream_failed": 'orange',
-    "skipped": 'pink',
-    "scheduled": 'tan',
+    "queued": "gray",
+    "running": "lime",
+    "success": "#0000FF",
+    "failed": "red",
+    "up_for_retry": "gold",
+    "up_for_reschedule": "turquoise",
+    "upstream_failed": "orange",
+    "skipped": "pink",
+    "scheduled": "tan",
 }
+
+
+def pod_mutation_hook(pod: V1Pod):
```

Comment:
I suspect because the upstream code lives outside the repo, and this is how you got things to work. If so I would normally say to import the upstream code, and modify the bits you want to change. But all of the k8s code for 2.0+ has gone under a lot of changes that I haven't been able to review yet, and a lot of work here will likely break when I upgrade.

I don't like this logic being put at a global place, which may require restarts/redeploys every time you need to modify something.

Reply:
This is how the `pod_mutation_hook` is supposed to be defined, as per the airflow documentation: https://airflow.apache.org/docs/apache-airflow/stable/kubernetes.html#pod-mutation-hook

Reply:
Ah, unfortunate. This effects all pods